quiz-template.html

```diff
@@ -1,148 +1,185 @@
-		categoryName = "Any"
-		description = "Hm maybe you should retake the quiz? :3"	
-	}
-}
-
-/* TODO: 
-- Check whether the user answered all questions.
-- Create a button to reset the quiz. 
-*/
-function displayResult(){
-	grade();
-	generateCategory();
-	hideHome();
-	document.write("<font face = 'Arial'><center>Congratulations! You are in the </font>" + "<br><font size = '14px'>" + categoryName + "</font><br>" + "<font face = 'Arial'>category!" + "<br>" + description + "</center></font>");
-}
-</script>
-
-<body bgcolor="#FFFFFF">
-
-<!-- Home -->
-<div id = "home">
-
-<!-- Form -->
-<form method="POST" name="quiz">
-
-<p><font face="Arial"><big><big>Quiz Title</big></big></font></p>
-
-<!-- Questions. NOTE: To avoid ties, choose 4, 7, 10, etc. questions -->
-<div class="questions">
-Question #1
-</div>
-
-<div class="answers">
-<input type ="radio" value ="x" name="question1"> a) Description <br>
-<input type ="radio" value ="y" name="question1"> b) Description <br>
-<input type ="radio" value ="z" name="question1"> c) Description <br>
-
-</div>
-
-<br>
-
-<div class="questions">
-Question #2
-</div>
-
-<div class="answers">
-<input type ="radio" value ="x" name="question2"> a) Description <br> 
-<input type ="radio" value ="y" name="question2"> b) Description <br>
-<input type ="radio" value ="z" name="question2"> c) Description <br>
-</div>
-
-<br>
-
-<div class="questions"">
-Question #3
-</div>
-
-<div class="answers">
-<input type ="radio" value ="x" name="question3"> a) Description <br>
-<input type ="radio" value ="y" name="question3"> b) Description <br>
-<input type ="radio" value ="z" name="question3"> c) Description <br> 
-</div>
-
-<br>
-
-</form>
-
-<!-- Submit button -->
-<form>
-<input type = "button" value="Submit!" name ="submit" onClick="displayResult()">
-</form>
-
-</div> 
+          }
+        }
+      }
+
+      /* Generates the text to appear in the results */
+      function generateCategory() {
+        if (xTotal > Math.max(yTotal, zTotal)) { // category x
+          categoryName = "X";
+          categoryDescription = "Supplementary information for x :3";
+        } else if (yTotal > Math.max(xTotal, zTotal)) { // category y
+          categoryName = "Y";
+          categoryDescription = "Supplementary information for y :3";
+        } else if (zTotal > Math.max(xTotal, yTotal)) { // catgory z
+          categoryName = "Z";
+          categoryDescription = "Supplementary information for z :3";
+        } else { // if ties are possible or if no questions are answered
+          categoryName = "Any";
+          categoryDescription = "Hm maybe you should retake the quiz? :3";
+        }
+      }
+
+      /* TODO: 
+         - Check whether the user answered all questions
+           for browsers that don't support the "required" attribute.
+       */
+      function displayResult() {
+        grade();
+        generateCategory();
+        hideMain();
+
+        // fill in fields

```

Comment:
These aren't in a separate function because this is what the function is for, according to its name.